CMakeLists.txt

```
@@ -52,14 +52,21 @@ include(CheckCXXSourceCompiles)
 include(CMakeDependentOption)
 include(CMakePushCheckState)
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 option(ENABLE_CLANG_TIDY "Enable clang-tidy")
 if (ENABLE_CLANG_TIDY)
     find_program(CLANG_TIDY_BIN NAMES "clang-tidy")
     set(CMAKE_C_CLANG_TIDY "${CLANG_TIDY_BIN}")
     set(CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_BIN}")
 endif()
 
+if (NOT CMAKE_BUILD_TYPE STREQUAL "Release")
+  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
+      add_compile_options(-fsanitize=address -fno-omit-frame-pointer)
+      add_link_options(-fsanitize=address)
+  endif()
+endif()
+
 function(add_cxx_warning WARNING)
```

```suggestion
endif()

function(add_cxx_warning WARNING)
```

Users should configure with `-DCMAKE_CXX_FLAGS=-fsanitize=address -DCMAKE_EXE_LINKER_FLAGS=-fsantitize=address -DCMAKE_SHARED_LINKER_FLAGS=-fsanitize=address`.

Or we could make it available as an extra option, e.g. `iSLE_ASAN`.

Building a release binary with sanitizing is always useful.

---

I need ASAN somehow so iSLE_ASAN it is.